packages/@aws-cdk/aws-iam/lib/policy-document.ts

```diff
@@ -20,13 +20,40 @@ export class PolicyDocument extends cdk.Token {
       return undefined;
     }
 
-    const doc = this.baseDocument || { };
-    doc.Statement = doc.Statement || [ ];
-    doc.Version = doc.Version || '2012-10-17';
-    doc.Statement = doc.Statement.concat(this.statements);
+    const doc = {
+      ...this.baseDocument,
+      Statement: (this.baseDocument.Statement || []).concat(this.statements),
```

Comment:
This concatenation happens at construction time - when `this.statements` is guaranteed to be empty. You need to wrap this in a `Token` so it is done at synthesis time.

Reply:
This is in `resolve()` not `constructor()`, it was like that before.

https://github.com/awslabs/aws-cdk/blob/b61707896582373c4526129945ddaac20055f313/packages/%40aws-cdk/aws-iam/lib/policy-document.ts#L23-L27